src/apk.c

```diff
@@ -575,7 +575,7 @@ int main(int argc, char **argv)
 		for (i = 0; optgroups[i]; i++) {
 			r = optgroups[i]->parse(ctx, &dbopts, p, optarg);
 			if (r == 0) break;
-			if (r == -EINVAL) {
+			if (r == -EPERM) {
```

Comment:
No applet ever returns `EPERM`. It would be then better to revert  5da4dec2a23e35153075838c4587c07dd20459f9 but it was supposed fix --help --verbose. This patch breaks again that case. Something different seems to be needed for proper fix.